### PR TITLE
Da clownana update! Clownana rework, global hOnKeRbLaSt & monkey synergy.

### PR DESCRIPTION
**Pull request by Krysonism**
## About The Pull Request
Since maintainer have expressed dissatisfaction with the state of clownanans, i've taken it upon myself to not only reduce their lag generation potential but to also make them more interesting.

All simplemob clowns now:
Are able to speak and understand the monkey language.
Are able to hit banana bunches to initiate the explosive ripening process.

The changes to clownananas speciecially are as follow;
Clownanas are able to rustle every 10 seconds to spawn 3 peels.
Clownanas have a speed of -1 rather than  -10.
Clownanas gain the ability to spawn a banana bunch every minute.

Banana bunches are food items imbued with extra banana juice and monkey energy. 

Hitting a banana bunch as a simplemob clown causes a honkerblast after a 3 seconds delay.

I've implemented a global honkerblast proc for future use, the honkerblast effects are similiar and inspired by the mech weapon, but less severe at light intensity and more severe at high intensity.

## Why It's Good For The Game
bild

Maintainers are unhappy with the current state of the mob, and while the mob is popular and enjoyed by many players it is currently quite poorly designed and implemented by a novice coder.

These changes increase the strategic depth of playing this particular mob and rewards more active input from the player.

## Changelog

:cl:
balance: Clownanas are slower, but still faster than humans.
balance: Clownanas no longer passively drop bananas, but gain an action that lets them drop three peels with a cooldown of 10 seconds.
balanance: Clown simple mobs now speak & understand monkey.
balance  Clowananas now get another new ability: Drop a monkey energy infused banana bunch, hit it to unleash a honkerblast!
/:cl:

### CODE REVIEW

**Comment by Farquaar** [comment]
Boooo
There are few greater cytology-related pleasures than unleashing your hard-grown clownanas into the escape shuttle hallway
Especially it's a multi-Z map and the floor is full of holes

File: code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm [hunk 1]
```diff
@@ -94,9 +85,118 @@
...
+	playsound(src, 'sound/effects/fuse.ogg', 80)
+
+	animate(src, time = 1, pixel_z = 12, easing = ELASTIC_EASING)
+	animate(time = 1, pixel_z = 0, easing = BOUNCE_EASING)
+	addtimer(CALLBACK(src, .proc/explosive_ripening), 3 SECONDS)
+	for(var/i in 1 to 32)
+		animate(color = (i % 2) ? "#ffffff": "#ff6739", time = 1, easing = QUAD_EASING)
+
+/obj/item/food/grown/banana/bunch/proc/explosive_ripening()
+	honkerblast(src, light_range = 3, medium_range = 1)
```
Comment (ATH1909) [inline code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm:194]: you have to wait 3 seconds for an area of effect this small?

File: code/__HELPERS/honkerblast.dm [hunk 2]
```diff
@@ -0,0 +1,47 @@
+///Unleashes a honkerblast similar to the honkmech weapon, but with more granular control.
```
Comment (ATH1909) [inline code/__HELPERS/honkerblast.dm:1]: why not make the honkmech weapon use this proc, then?

File: code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm (hunk 1 above)
Comment (Krysonism) [reply code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm:194]: yes

File: code/__HELPERS/honkerblast.dm (hunk 2 above)
Comment (Krysonism) [reply code/__HELPERS/honkerblast.dm:1]: Well i didn't want to get into the honk mech balance discussion on this PR.

File: code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm [hunk 3]
```diff
@@ -94,9 +85,118 @@
 	emote_see = list("honks", "bites into the banana", "plucks a banana off its head", "photosynthesizes")
 	maxHealth = 120
 	health = 120
-	speed = -10
+	speed = -1
 	loot = list(/obj/item/clothing/mask/gas/clown_hat, /obj/effect/gibspawner/human, /obj/item/soap, /obj/item/seeds/banana)
-	banana_time = 20
+	//Our peel dropping ability
```
Comment (SuperNovaa41) [inline code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm:90]:
```suggestion
	///Our peel dropping ability
```

File: code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm [hunk 4]
```diff
@@ -94,9 +85,118 @@
...
+	banana_bunch.Grant(src)
+
+///drops peels around the mob when activated
+/datum/action/cooldown/rustle
+	name = "Rustle"
+	desc = "Shake loose a few banana peels."
+	///which type of peel to spawn
+	var/banana_type = /obj/item/grown/bananapeel
+	///How many peels to spawn
+	var/peel_amount = 3
```
Comment (SuperNovaa41) [inline code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm:109]: these new vars should be below all of the preexisting ones

File: code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm [hunk 5]
```diff
@@ -94,9 +85,118 @@
...
+	for(var/turf/adjacent_turf in RANGE_TURFS(1, owner.loc))
+		if(adjacent_turf == owner.loc || !owner.CanReach(adjacent_turf) || !isopenturf(adjacent_turf))
+			continue
+		reachable_turfs += adjacent_turf
+
+	var/peels_to_spawn = min(peel_amount, reachable_turfs.len)
+	for(var/i in 1 to peels_to_spawn)
+		new banana_type(pick_n_take(reachable_turfs))
+	playsound(owner, 'sound/creatures/clown/clownana_rustle.ogg', 100)
+	animate(owner, time =1, pixel_x = 6, easing = CUBIC_EASING | EASE_OUT)
```
Comment (SuperNovaa41) [inline code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm:127]:
```suggestion
	animate(owner, time = 1, pixel_x = 6, easing = CUBIC_EASING | EASE_OUT)
```
https://github.com/tgstation/tgstation/blob/master/.github/guides/STYLE.md#spacing

File: code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm [hunk 6]
```diff
@@ -94,9 +85,118 @@
...
+	animate(time = 2, pixel_x = -8, easing = CUBIC_EASING)
+	animate(time =  1, pixel_x = 0, easing = CUBIC_EASING | EASE_IN)
```
Comment (SuperNovaa41) [inline code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm:129]:
```suggestion
	animate(time = 1, pixel_x = 0, easing = CUBIC_EASING | EASE_IN)
```

File: code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm [hunk 7]
```diff
@@ -94,9 +85,118 @@
...
+		return
+	activating = TRUE
+	if(!do_after(owner, 1 SECONDS))
+		activating = FALSE
+		return
+	playsound(owner, 'sound/creatures/clown/hehe.ogg', 100)
+	if(!do_after(owner, 1 SECONDS))
+		activating = FALSE
+		return
+	activating = FALSE
```
Comment (SuperNovaa41) [inline code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm:160]:
```suggestion
	playsound(owner, 'sound/creatures/clown/hehe.ogg', 100)
    activating = FALSE
	if(!do_after(owner, 1 SECONDS))
		return
```

Comment (Krysonism) [reply code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm:159]: Setting activating to false before the do_after will allow the user to spam the button and create multiple bunches,

**Comment by LemonInTheDark** [comment]
> Boooo There are few greater cytology-related pleasures than unleashing your hard-grown clownanas into the escape shuttle hallway Especially it's a multi-Z map and the floor is full of holes

Classic Stats

**Comment by ATH1909** [comment]
lmao

**Review by Twaticus — APPROVED** [review]

**Comment by jughu** [comment]
You could make the bananas like the honk shoes where they dispear once they are slipped on it will not help alot but it will partly deal with the issue of them littering it everywherr after having slipped somoene

File: code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm [hunk 8]
```diff
@@ -94,9 +84,87 @@
...
+	var/datum/action/cooldown/rustle/banana_rustle
+	///Our banana bunch spawning ability
+	var/datum/action/cooldown/exquisite_bunch/banana_bunch
+
+/mob/living/simple_animal/hostile/retaliate/clown/banana/Initialize(mapload)
+	. = ..()
+	banana_rustle = new()
+	banana_rustle.Grant(src)
+	banana_bunch = new()
+	banana_bunch.Grant(src)
```
Comment (LemonInTheDark) [inline code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm:99]: it's not technically necessary, but could you null your two action refs when the clown Destroy()'s

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]
Few small things

File: code/modules/hydroponics/grown/banana.dm [hunk 9]
```diff
@@ -144,3 +144,34 @@
...
+/obj/item/food/grown/banana/bunch/proc/start_ripening()
+	if(is_ripening)
+		return
+	playsound(src, 'sound/effects/fuse.ogg', 80)
+
+	animate(src, time = 1, pixel_z = 12, easing = ELASTIC_EASING)
+	animate(time = 1, pixel_z = 0, easing = BOUNCE_EASING)
+	addtimer(CALLBACK(src, .proc/explosive_ripening), 3 SECONDS)
+	for(var/i in 1 to 32)
+		animate(color = (i % 2) ? "#ffffff": "#ff6739", time = 1, easing = QUAD_EASING)
```
[the remark this answers is not in the document]
Comment (Krysonism) [reply code/modules/hydroponics/grown/banana.dm:169]: Yes

File: code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm (hunk 8 above)
Comment (Krysonism) [reply code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm:99]: sure

**Comment by Krysonism** [comment]
conflict moment.

File: code/modules/hydroponics/grown/banana.dm (hunk 9 above)
[the remark this answers is not in the document]
Comment (LemonInTheDark) [reply code/modules/hydroponics/grown/banana.dm:169]: Why not make the animate into an infinite loop rather then timing it out like this

[the remark this answers is not in the document]
Comment (Krysonism) [reply code/modules/hydroponics/grown/banana.dm:169]: i tried for(var/i in 1 to INFINITY) and it froze the game.

[the remark this answers is not in the document]
Comment (Krysonism) [reply code/modules/hydroponics/grown/banana.dm:169]: Making the loop infinite freezes the game.

[the remark this answers is not in the document]
Comment (LemonInTheDark) [reply code/modules/hydroponics/grown/banana.dm:169]: I uh, meant loop = -1. V based tho

**Comment by LemonInTheDark** [comment]
@Krysonism you alive? I'm up for merging this as is if you're too zonked

**Comment by LemonInTheDark** [comment]
He's busy for reasons, letting this sit till he's up and at em again. if I get bored I'll just force push the changes and merge, but I've not seen any recent stuff like this, so no reason to as of now

File: code/modules/hydroponics/grown/banana.dm (hunk 9 above)
[the remark this answers is not in the document]
Comment (LemonInTheDark) [reply code/modules/hydroponics/grown/banana.dm:169]: doesn't work, curse you byond

**Pull request merged**